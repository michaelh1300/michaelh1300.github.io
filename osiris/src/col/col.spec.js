import { shallowMount } from '@vue/test-utils';
import Col from './col.vue';

describe('Col', () => {
  let wrapper;
  describe('when pass no prop', () => {
    beforeEach(() => {
      wrapper = shallowMount(Col);
    });

    it('should match with the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('should have default classList', () => {
      const expected = ['o-col-24', 'o-col-offset-0'];
      expect(wrapper.vm.classList).toEqual(expected);
    });
  });

  describe('when pass some props', () => {
    const props = {
      span: 1,
      xs: 2,
      sm: 3,
      md: 4,
      lg: 5,
      xl: 6,
      offset: 7,
    };

    describe(`and the props are ${JSON.stringify(props)}`, () => {
      beforeEach(() => {
        wrapper = shallowMount(Col, {
          propsData: {
            ...props,
          },
        });
      });

      it('should match with the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

      it('should have exactly expected classList', () => {
        const expected = [
          'o-col-1',
          'o-col-offset-7',
          'o-col-xs-2',
          'o-col-sm-3',
          'o-col-md-4',
          'o-col-lg-5',
          'o-col-xl-6',
        ];

        expect(wrapper.vm.classList).toEqual(expected);
      });
    });

    describe('and the props are another tag like section', () => {
      beforeEach(() => {
        wrapper = shallowMount(Col, {
          propsData: {
            tag: 'section',
          },
        });
      });

      it('should match with the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
});
