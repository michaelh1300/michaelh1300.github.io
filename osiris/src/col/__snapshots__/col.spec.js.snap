// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col when pass no prop should match with the snapshot 1`] = `<div class="o-col-24 o-col-offset-0" style="padding-left: 0px; padding-right: 0px;"></div>`;

exports[`Col when pass some props and the props are {"span":1,"xs":2,"sm":3,"md":4,"lg":5,"xl":6,"offset":7} should match with the snapshot 1`] = `<div class="o-col-1 o-col-offset-7 o-col-xs-2 o-col-sm-3 o-col-md-4 o-col-lg-5 o-col-xl-6" style="padding-left: 0px; padding-right: 0px;"></div>`;

exports[`Col when pass some props and the props are another tag like section should match with the snapshot 1`] = `<section class="o-col-24 o-col-offset-0" style="padding-left: 0px; padding-right: 0px;"></section>`;
